lib/flagsmith/sdk/models/flags.rb

```diff
@@ -119,16 +120,27 @@ def feature_value(feature_name)
       end
       alias get_feature_value feature_value
 
+      def get_flag_from_offline_handler(key)
+        @offline_handler.environment.feature_states.each do |feature_state|
+          return Flag.from_feature_state_model(feature_state, nil) if key == Flagsmith::Flags::Collection.normalize_key(feature_state.feature.name)
+        end
+        raise Flagsmith::Flags::NotFound,
+              "Feature does not exist: #{key}, offline_handler did not find a flag in this case."
+      end
+
       # Get a specific flag given the feature name.
       # :param feature_name: the name of the feature to retrieve the flag for.
       # :return: BaseFlag object.
       # :raises FlagsmithClientError: if feature doesn't exist
       def get_flag(feature_name)
         key = Flagsmith::Flags::Collection.normalize_key(feature_name)
+
         flag = flags.fetch(key)
         @analytics_processor.track_feature(flag.feature_name) if @analytics_processor && flag.feature_id
         flag
       rescue KeyError
+        return get_flag_from_offline_handler(key) if @offline_handler
```

Comment:
Ah, I guess this answers my previous question then. ... I feel like this also belongs in the python client though?

Reply:
From what I could follow from the Python client, the structure of the code was different and didn't necessitate sending it down into the class. This is off memory though. I remember disliking having to send in the offline_handler into this random class, but at least it does solve the problem.